data/transactions/logic/assembler.go

```diff
@@ -1531,6 +1534,38 @@ func (ops *OpStream) trackStack(args StackTypes, returns StackTypes, instruction
 	}
 }
 
+func processFields(ops *OpStream, fields []string) (current, next []string) {
```

Comment:
comment to describe processFields. be very clear about whether `fields` is modified and whether any of the output slices share underlying arrays with each other or `fields`.

and should it be a method of OpStream?